File: study-ai-myapp-fimaly/backend/src/main/java/com/familytree/model/Member.java
package com.familytree.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import jakarta.persistence.*;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "members")
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "family_id", nullable = false)
    private Long familyId;
    
    @Column(name = "name", nullable = false)
    private String name;
    
    @Column(name = "gender")
    private String gender;
    
    @Column(name = "birth_date")
    @Temporal(TemporalType.DATE)
    private Date birthDate;
    
    @Column(name = "death_date")
    @Temporal(TemporalType.DATE)
    private Date deathDate;
    
    @Column(name = "photo")
    private String photo;
    
    @Column(name = "details", columnDefinition = "TEXT")
    private String details;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = new Date();
    }
}

File: study-ai-myapp-fimaly/frontend/android/lib/models/member.dart
class Member {
  final int id;
  final int familyId;
  final String name;
  final String gender;
  final DateTime? birthDate;
  final DateTime? deathDate;
  final String? photo;
  final String? details;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Member({
    required this.id,
    required this.familyId,
    required this.name,
    required this.gender,
    this.birthDate,
    this.deathDate,
    this.photo,
    this.details,
    required this.createdAt,
    this.updatedAt,
  });

  factory Member.fromJson(Map<String, dynamic> json) {
    return Member(
      id: json['id'],
      familyId: json['familyId'],
      name: json['name'],
      gender: json['gender'],
      birthDate: json['birthDate'] != null ? DateTime.parse(json['birthDate']) : null,
      deathDate: json['deathDate'] != null ? DateTime.parse(json['deathDate']) : null,
      photo: json['photo'],
      details: json['details'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'familyId': familyId,
      'name': name,
      'gender': gender,
      'birthDate': birthDate?.toIso8601String(),
      'deathDate': deathDate?.toIso8601String(),
      'photo': photo,
      'details': details,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}